### PR TITLE
rbd: drop unnecessary using declaration, etc

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

File: src/krbd.cc [hunk 1]
```diff
@@ -40,7 +40,7 @@
 #include <blkid/blkid.h>
 #include <libudev.h>
 
-using namespace std;
+// using namespace std;
```
Comment (joscollin) [inline src/krbd.cc]: Drop commented line

File: src/krbd.cc [hunk 2]
```diff
@@ -133,10 +133,13 @@ static int build_map_buf(CephContext *cct, const char *pool, const char *image,
...
-  if (cct->_conf->auth_client_required != "none") {
+  auto auth_client_required =
+    cct->_conf->get_val<std::string>("auth_client_required");
+  if (auth_client_required != "none") {
     r = keyring.from_ceph_context(cct);
-    if (r == -ENOENT && !(cct->_conf->keyfile.length() ||
-                          cct->_conf->key.length()))
+    auto keyfile = cct->_conf->get_val<std::string>("keyfile");
+    auto key = cct->_conf->get_val<std::string>("key");
+    if (r == -ENOENT && !(keyfile.length() || key.length()))
```
Comment (dillaman) [inline src/krbd.cc]: Nit: if you are cleaning this up, might as well change ```!(<size_t> || <size_t>)``` to be ```keyfile.empty() && key.empty()``` for clarity

File: src/krbd.cc [hunk 3]
```diff
@@ -777,6 +780,7 @@ extern "C" void krbd_destroy(struct krbd_ctx *ctx)
   udev_unref(ctx->udev);
 
   delete ctx;
+  ctx = nullptr;
```
Comment (dillaman) [inline src/krbd.cc]: Nit: unnecessary change, this is a local variable that goes out-of-scope

**Comment by shinobu-x** [comment]
@dillaman Please check when you get a chance.

File: src/krbd.cc (hunk 1 above)
Comment (dillaman) [reply src/krbd.cc]: ... this still needs to be addressed

**Comment by dillaman** [comment]
@shinobu-x I don't see any review comments being addressed. Also, can you please change your commit subject to match the PR's title? Thanks.
